front/lib/api/assistant/actions/conversation/include_file.ts

```diff
@@ -0,0 +1,171 @@
+import type {
+  ContentFragmentType,
+  ConversationType,
+  FunctionCallType,
+  FunctionMessageTypeModel,
+  ModelConfigurationType,
+  ModelId,
+  SupportedContentFragmentType,
+} from "@dust-tt/types";
+import { CoreAPI } from "@dust-tt/types";
+import {
+  assertNever,
+  BaseAction,
+  isContentFragmentType,
+  isSupportedImageContentType,
+  isTextContent,
+} from "@dust-tt/types";
+
+import config from "@app/lib/api/config";
+import { renderContentFragmentForModel } from "@app/lib/resources/content_fragment_resource";
+import logger from "@app/logger/logger";
+
+export function isConversationIncludableFileContentType(
+  contentType: SupportedContentFragmentType
+): boolean {
+  if (isSupportedImageContentType(contentType)) {
+    return false;
+  }
+  // For now we only allow including text files.
+  switch (contentType) {
+    case "application/msword":
+    case "application/vnd.openxmlformats-officedocument.wordprocessingml.document":
+    case "application/pdf":
+    case "text/markdown":
+    case "text/plain":
+    case "dust-application/slack":
+      return true;
+
+    case "text/comma-separated-values":
+    case "text/csv":
+    case "text/tab-separated-values":
+    case "text/tsv":
+      return false;
+    default:
+      assertNever(contentType);
+  }
+}
+
+interface ConversationIncludeFilesActionBlob {
+  id: ModelId;
+  agentMessageId: ModelId;
+  params: {
+    fileId: string;
+  };
+  functionCallId: string | null;
+  functionCallName: string | null;
+  step: number;
+}
+
+const CONTEXT_SIZE_DIVISOR_FOR_INCLUDE = 4;
+
+export class ConversationIncludeFilesAction extends BaseAction {
+  readonly agentMessageId: ModelId;
+  readonly params: {
+    fileId: string;
+  };
+  readonly contentFragments: null[] = [];
+  readonly functionCallId: string | null;
+  readonly functionCallName: string | null;
+  readonly step: number = -1;
+  readonly type = "conversation_include_files_action";
+
+  constructor(blob: ConversationIncludeFilesActionBlob) {
+    super(blob.id, "conversation_include_files_action");
+
+    this.agentMessageId = blob.agentMessageId;
+    this.params = blob.params;
+    this.functionCallId = blob.functionCallId;
+    this.functionCallName = blob.functionCallName;
+    this.step = blob.step;
+  }
+
+  renderForFunctionCall(): FunctionCallType {
+    return {
+      id: this.functionCallId ?? `call_${this.id.toString()}`,
+      name: this.functionCallName ?? "include_conversation_files",
```

Comment:
files => file

Reply:
Yes will rename it everywhere :+1: (predates decision to go for just one file)